File: largest_odd_number_in_string.py
def largestOddNumber(num:str) -> str:
    for i in range(len(num)-1,-1,-1):
        if int(num[i]) % 2 == 1:
            return num[:i + 1]
    return ""

print(largestOddNumber("23232")) #Output = 2323

File: largest_odd_number_in_string.js
const largestOddNumber = function (num) {
  for (let i = num.length - 1; i >= 0; i--) {
    if (num[i] % 2 === 1) return num.slice(0, i + 1);
  }
  return "";
};

console.log(largestOddNumber("42021")); // OutPut: 42021
